sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md

```diff
@@ -81,57 +81,55 @@ directive:
-      $.Permissions.properties.storage.items['x-ms-enum']['name'] = 'StoragePermission';
-      $.Resource['x-ms-client-name'] = 'KeyVaultResourceData';
+      $.Permissions.properties.keys.items['x-ms-enum']['name'] = 'IdentityAccessKeyPermission';
+      $.Permissions.properties.secrets.items['x-ms-enum']['name'] = 'IdentityAccessSecretPermission';
+      $.Permissions.properties.certificates.items['x-ms-enum']['name'] = 'IdentityAccessCertificatePermission';
+      $.Permissions.properties.storage.items['x-ms-enum']['name'] = 'IdentityAccessStoragePermission';
       $.IPRule.properties.value['x-ms-client-name'] = 'AddressRange';
       $.IPRule['x-ms-client-name'] = 'VaultIPRule';
+      $.VirtualNetworkRule['x-ms-client-name'] = 'VaultVirtualNetworkRule';
       $.DeletedVaultProperties.properties.vaultId['x-ms-format'] = 'arm-id';
       $.Vault['x-csharp-usage'] = 'model,input,output';
       $.VaultProperties.properties.createMode['x-ms-enum']['name'] = 'VaultCreateMode';
       $.VaultProperties.properties.networkAcls['x-ms-client-name'] = 'NetworkRuleSet';
       $.VaultPatchProperties.properties.createMode['x-ms-enum']['name'] = 'VaultPatchMode';
       $.VaultPatchProperties.properties.networkAcls['x-ms-client-name'] = 'NetworkRuleSet';
-      $.Resource.properties.location['x-ms-format'] = 'azure-location';
       $.PrivateEndpointConnectionItem.properties.etag['x-ms-format'] = 'etag';
       $.PrivateEndpointConnection.properties.etag['x-ms-format'] = 'etag';
-      $.PrivateLinkServiceConnectionState.properties.actionsRequired['x-ms-enum']['name'] = 'ActionsRequiredMessage';
+      $.PrivateLinkServiceConnectionState.properties.actionsRequired['x-ms-enum']['name'] = 'VaultActionsRequiredMessage';
       $.VaultCheckNameAvailabilityParameters.properties.type['x-ms-format'] = 'resource-type';
       $.VaultCheckNameAvailabilityParameters['x-ms-client-name'] = 'VaultNameAvailabilityParameters';
-      $.NetworkRuleSet.properties.bypass['x-ms-enum']['name'] = 'NetworkRuleBypassOption';
+      $.NetworkRuleSet.properties.bypass['x-ms-enum']['name'] = 'VaultNetworkRuleBypassOption';
       $.NetworkRuleSet['x-ms-client-name'] = 'VaultNetworkRuleSet';
       $.AccessPolicyEntry['x-ms-client-name'] = 'VaultAccessPolicy';
-  - rename-model:
-      from: MHSMIPRule
-      to: MhsmIPRule
-  - rename-model:
-      from: Attributes
-      to: BaseAttributes
   - rename-model:
       from: Permissions
       to: IdentityAccessPermissions
+  - rename-model:
+      from: MHSMIPRule
+      to: MhsmIPRule
   - rename-model:
       from: MHSMPrivateLinkResource
       to: MHSMPrivateLinkResourceData
-  - rename-model:
-      from: PrivateLinkResource
-      to: PrivateLinkResourceData
   - rename-model:
       from: MHSMPrivateEndpointConnectionItem
       to: MHSMPrivateEndpointConnectionItemData
+  - rename-model:
+      from: PrivateLinkResource
+      to: VaultPrivateLinkResourceData
```

Comment:
You could try to remove these three lines, our common shcema renaming mechanism will rename this to `KeyVaultPrivateLinkResourceData` which aligns with other classes (like `KeyVaultPrivateLinkResourceListResult`)

Reply:
The suffix `Resource` can be renamed to `ResourceData`, and here just make me realize that naming of the vault service is not consistency, so change all `KeyVault` prefix to `Vault` prefix that to be very clear of the 2 sub-provider for KeyVault: Vault and ManagedHsm.